frame/assets/src/functions.rs

```diff
@@ -661,8 +661,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				status: AssetStatus::Live,
 			},
 		);
+		ensure!(T::CallbackHandle::created(&id, &owner).is_ok(), Error::<T, I>::CallbackFailed);
```

Comment:
This relies on transactional stroage since the insert above happens after the failure here.

Reply:
you can use `require_transactional` to annotate it https://github.com/paritytech/substrate/blob/959c5b636f2536b883f64e13591aa61825a558f5/frame/support/src/lib.rs#L657